trace/amqp/amqp.go

```diff
@@ -134,11 +134,11 @@ func (tc *TracedPublisher) Publish(ctx context.Context, msg *Message) error {
 	p.Headers[correlation.HeaderID] = correlation.IDFromContext(ctx)
 
 	err = tc.ch.Publish(tc.exc, "", false, false, p)
+	defer trace.SpanComplete(sp, err)
```

Comment:
No need to defer here.
```suggestion
	trace.SpanComplete(sp, err)
```

Reply:
Hi, thanks for review. 

defer in this cases makes the code more readable and it is suitable for the purpose of defer, although as you mentioned there is performance cost. However, as Brad Fitzpatrick said in Go 1.13, real improvement has been done for defer performance cost and in Go 1.14 nearly there will be no overhead.

https://twitter.com/bradfitz/status/1184547163235184640

Reply:
But in the code blocks you commented, thanks to defer stmt, we only gain few lines of code and that might not be needed for that case. I am going to remove them now.